EIPS/eip-draft_deposit_contract_snapshot_interface.md

```diff
@@ -0,0 +1,282 @@
+
+This EIP solves these issues by defining a standard format for transmitting the deposit contract Merkle tree in a compressed form to newly syncing nodes during weak subjectivity sync.
+
+## Specification
+Consensus clients MAY continue to implement the deposit Merkle tree however they choose. However, when transmitting the tree to newly syncing nodes, clients MUST use the following format:
+
+```
+DepositTreeSnapshot {
+    finalized: Vector[Hash32],
+    deposits:  uint64,
+    eth1_block_hash: Hash32,
+}
+```
+
+Where the hashes in the `finalized` vector are defined in the [Deposit Finalization Flow](#deposit-finalization-flow) section below, `deposits` is the number of deposits stored in the snapshot, and `eth1_block_hash` is the hash of the eth1 block containing the highest index deposit stored in the snapshot. Consensus clients MUST make this structure available via the Beacon Node API endpoint `/eth/v2/beacon/deposit_snapshot`.
+
+#### Deposit Finalization Flow
+
+During deposit processing, the beacon chain requires deposits to be submitted along with a Merkle path to the deposit root. This is required exactly once for each deposit. When a deposit has been processed by the beacon chain and the [deposit finalization conditions](#deposit-finalization-conditions) have been met, many of the hashes along the path to the deposit root will never be required again to construct Merkle proofs on chain. These unnecessary hashes MAY be pruned to save space. The image below illustrates the evolution of the deposit Merkle tree under this process alongside the corresponding `DepositTreeSnapshot` as new deposits are added and older deposits become finalized:
+
+![Deposit Tree Evolution](../assets/eip-draft_deposit_contract_snapshot_interface/deposit_tree_evolution.png)
+
+## Rationale
+
+The format in this specification was chosen to achieve several goals simultaneously:
+
+1. Enable reconstruction of the deposit contract Merkle tree under the adoption of [EIP-4444](./eip-4444.md)
+2. Avoid requiring consensus nodes to retain more deposits than necessary to fully participate in consensus
+3. Simplicity of implementation (see [Reference Implementation](#reference-implementation) section)
+4. Increase speed of weak subjectivity sync
+5. Compatibility with existing implementations of this mechanism (see [Existing Implementations](#existing-implementations))
+
+#### Why not Reconstruct the Tree Directly from the Deposit Contract?
+
+The deposit contract can only provide the tree at the head of the chain. Because the beacon chain's view of the deposit contract lags behind the eth1 chain by `ETH1_FOLLOW_DISTANCE`, there are almost always deposits which haven't yet been included in the chain that need proofs constructed from an earlier version of the tree than exists at the head.
```

Comment:
We shouldn't be using the "eth1" terminology anywhere in our specifications (including in supplementary material like the rationale).  Execution layer, execution client, execution engine, etc. are all reasonable depending on context.  Same comment a number of other places in the supplemental sections of this document.

Reply:
What is the long term plan for updating the `BeaconState` spec to correctly label things as execution/consensus instead of eth1/eth2?